analysis/descriptives/venn_diagram.R

```diff
@@ -223,6 +223,17 @@ venn_output <- function(cohort_name){
   
   # Save summary file ----------------------------------------------------------
   
+  # Merge any cells <= 5 to the highest cell (excluding totals) -------------
+  
+  colnamesorder <- colnames(df)
+  a <- df[-c(1,9:12)]
+  a[] <- sapply(a, as.numeric)
+  idx <- cbind(seq(nrow(a)), max.col(a))
+  a[idx] <- a[idx] + rowSums(a * (a <= 5))
+  is.na(a) <- a <= 5
+  df <- cbind(df[c(1,9:12)], a) 
```

Comment:
Hi @yinghuiwei - I don't think the totals need to be redacted as they will never be less than 5. The totals remain exactly the same, because the way this is coded is that if any numbers <= 5 are present, then that total is then added to the largest category of that row (as Angela requested in the meeting where we discussed this). E.G. if there is just 1 count for "snomed_hes", this 1 is added to the "snomed_only" category (assuming this has the highest count) and the total remains exactly the same. We are effectively swapping numbers around. Please phone me on Slack if this is unclear :).

Reply:
This has now been resolved by removing four columns: total_*, which can be calculated externally by adding numbers from relevant columns 2-8 (only_* and the intersections).